src/parser/mod.rs

```diff
@@ -853,14 +853,22 @@ impl<'a> Parser<'a> {
 
         Ok(Statement::ReleaseSavepoint { name })
     }
-
+    pub fn parse_column_with_data_type(&mut self) -> Result<Expr, ParserError> {
+        self.expect_token(&Token::LBrace)?;
+        let column_with_data_type = self.parse_expr()?;
+        self.expect_token(&Token::RBrace)?;
+        Ok(column_with_data_type)
+    }
     /// Parse an expression prefix
     pub fn parse_prefix(&mut self) -> Result<Expr, ParserError> {
         // allow the dialect to override prefix parsing
         if let Some(prefix) = self.dialect.parse_prefix(self) {
             return prefix;
         }
-
+        // ClickHouse support using {column:Datatype} format in its creation of parametrised views
+        if dialect_of!(self is ClickHouseDialect) && self.peek_token().token == Token::LBrace {
+            return self.parse_column_with_data_type();
+        }
```

Comment:
Currently the parse logic for this feature is spread across this and the parse_infix function where one triggers the other indirectly, slightly harder to follow as a result - one suggestion to simplify this would be:
- The current function already includes a match onthe token type, so we can introduce a clause
   ```rust
   Token::LBRACE if dialect_of!(self is Clickhouse) => {
       self.prev_token();
       self.parse_column_with_data_type()
   }
    ```
- The current `parse_column_with_data_type` can parse the components directly - especially given that it does not seem correct for it to parse an inner expression. It becomes
   ```rust
   // consume LBRACE
   let name = self.parse_identifier()?;
   self.expect(Token::Colon)?;
   let data_type = self.parse_data_type()?;
   // consume RBRACE
   ```

Reply:
make sense to me.